src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_client.py

```diff
@@ -57,7 +57,15 @@ def _send(self, method, url, param=None, body=None, api_version=V1_0):
                 r = send_raw_request(self._cli_ctx, method, url, resource=self._resource, uri_parameters=param,
                                      body=body)
             except HTTPError as ex:
-                raise GraphError(ex.response.json()['error']['message'], ex.response) from ex
+                err = ex.response.json()['error']
+                code = err['code']
+                message = err['message']
+                graph_error_message = (
+                    f"({code}) {message}\n"
+                    f"Code: {code}\n"
+                    f"Message: {message}"
```

Comment:
Besides `code` and `message`, Azure Core also handles `target`, `details` and `innererror`.

Even though Graph error also contains similar properties, such as `error.details.target` and `error.innerError`:

```json
{
    "error": {
        "code": "CannotUpdateLockedServicePrincipalProperty",
        "message": "Property passwordCredentials is invalid.",
        "details": [
            {
                "code": "GenericError",
                "message": "Property passwordCredentials is invalid.",
                "target": "passwordCredentials",
                "blockedWord": "",
                "prefix": "",
                "suffix": ""
            }
        ],
        "innerError": {
            "date": "2024-09-20T08:01:41",
            "request-id": "fbbd7cdb-8181-4bcf-8bca-7ed896035ee2",
            "client-request-id": "fbbd7cdb-8181-4bcf-8bca-7ed896035ee2"
        }
    }
}
```

It may not be guaranteed that the error body of every API is the same, so I tend to only show the top level `error.code` and `error.message`, and not complicate things too much. The detailed information is in `--verbose` and `--debug` log after all.